### PR TITLE
QVAC-11763: add tests for sliding context

### PR DESCRIPTION
**Pull request by donriddo**
## 🎯 What problem does this PR solve?

- Adds comprehensive test coverage for the sliding context feature in llama.cpp LLM inference
- Ensures sliding context behavior is properly validated across different scenarios and edge cases, including cached follow-up messages

## 📝 How does it solve it?

- Adds a new integration test file `test/integration/sliding-context.test.js` with 10 test cases covering:
  - Basic sliding context activation when generation exceeds context limit
  - Context overflow error when sliding is disabled (`n_discarded=0`)
  - Multiple sliding activations during long generation sequences with small `n_discarded` values
  - Automatic clamping of oversized `n_discarded` values to fit available context space
  - Persistence of `n_discarded` setting across consecutive inference runs
  - Edge case handling with minimal `n_discarded=1` value
  - Cached follow-up messages that discard middle tokens to fit new messages
  - Cached follow-up behavior when discard window is exhausted
  - Cached follow-up overflow errors when sliding is disabled and context is full
  - Single prompt overflow detection at prefill stage

## 🧪 How was it tested?

- All 10 new test cases pass, validating sliding context behavior under various configurations
- Tests use a model (Llama-3.2-1B-Instruct-Q4_0) with constrained context size (256) to reliably trigger sliding behavior
- Tests verify both successful sliding activation and proper error handling when sliding is disabled
- Tests cover both generation-time sliding and cached follow-up scenarios

### CODE REVIEW

File: packages/qvac-lib-infer-onnx-tts/benchmarks/python-server/requirements.txt
Comment (gianni-cor) [inline packages/qvac-lib-infer-onnx-tts/benchmarks/python-server/requirements.txt:1]: @donriddo why changed this?

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/tool-calling.test.js [hunk 1]
```diff
@@ -172,6 +172,9 @@ async function createToolModel (modelVariant) {
       await model.unload().catch(() => {})
       await loader.close().catch(() => {})
       releaseLogger()
+      // Schedule a timer to keep the event loop alive briefly for C++ async cleanup
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/tool-calling.test.js]: i think this can be removed cause it doesn't fix the issue

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+  const discardCount = countDiscardLogs(logs)
+  t.is(
+    discardCount,
+    expectedSlides(SLIDE_PREDICT, 99999),
+    'slide count matches expected n_predict / clamped n_discarded'
+  )
+})
+
+// n_discarded=32, n_predict=SLIDE_PREDICT
+test('Sliding context persists across consecutive inference runs: total = 20 slides', {
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js]: i don't know what you wanted to test here,  but because you are not sending any "session" msg to the model, then the two runs are independent. So maybe can we remove this test?

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/cache-state-machine.test.js [hunk 3]
```diff
@@ -117,6 +117,9 @@ async function setupModel (t, overrides = {}) {
     await model.unload().catch(() => {})
     await loader.close().catch(() => {})
     releaseLogger()
+    // Schedule a timer to keep the event loop alive briefly for C++ async cleanup
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/cache-state-machine.test.js]: can be remove cause it doesn't fix the issue

File: packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+  }
+
+  t.is(countPrefillDiscardLogs(logs), 0, 'no prefill discard logs when n_discarded=0')
+
+  // sleep for 10 seconds to allow the model to cleanup
+  await new Promise(resolve => setTimeout(resolve, 10000))
+})
+
+// nTokens >= n_ct:> ContextOverflow before any sliding logic
+test('Single prompt exceeding context triggers overflow at prefill', {
```
Comment (gianni-cor) [inline packages/qvac-lib-infer-llamacpp-llm/test/integration/sliding-context.test.js]: can be removed cause already tested elsewhere

**Review by gianni-cor — CHANGES_REQUESTED** [review]

**Comment by gianni-cor** [comment]
@donriddo  skip multiInstance and MoE Integration tests on windows and leave a comment in the scripts saying they are temporarily disabled waiting for a new windows runner with gpu.

**Review by dev-nid — APPROVED** [review]

**Review by gianni-cor — APPROVED** [review]
